### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

**Comment by WyriHaximus** [comment]
LGTM :+1:

File: composer.json [hunk 1]
```diff
@@ -7,8 +7,18 @@
         "php": ">=5.4.0",
         "react/promise": "~2.0"
     },
+    "require-dev": {
+        "phpunit/phpunit": "~4.5"
```
Comment (clue) [inline composer.json]: Not sure if we ever reached a consensus on this..
*Personally*, I think it makes sense to explicitly define *all* dependencies - however this is something we've discussed in the past and have chosen to *not* do this for the other components.
As such, I'd recommend to keep this for a separate PR/discussion so that we can get this PR in earlier and am :-1: (on only this line) for now.

Comment (assertchris) [inline composer.json]: Super curious why this would be omitted. It reduces friction for new contributors and is ignored when this repository is consumed as a dependency...

Comment (clue) [inline composer.json]: The closest I could find is this: https://github.com/reactphp/child-process/pull/5#discussion_r13724410

IMHO this is still worth discussing, however this is somewhat unrelated to your other changes. I'd vote to open a (meta) issue on reactphp/react so that we can discuss and apply this for *all* components.

Comment (WyriHaximus) [inline composer.json]: :+1: on discussion it as I agree with @assertchris that it would make it easier for anyone to jump in

**Comment by assertchris** [comment]
Ok, I've removed PHPUnit as a dev dependency (same as in https://github.com/reactphp/whois/pull/9).

**Comment by assertchris** [comment]
@WyriHaximus ping

**Comment by r4j4h** [comment]
+1

**Comment by WyriHaximus** [comment]
@assertchris I have no powers in this place so pinging @cboden and @clue for you

**Comment by assertchris** [comment]
Just pinging here to see if anybody with merge rights wants to settle this...

File: tests/bootstrap.php [hunk 2]
```diff
@@ -1,7 +0,0 @@
-<?php
-
-$loader = @include __DIR__ . '/../vendor/autoload.php';
-if (!$loader) {
-    $loader = require __DIR__ . '/../../../../vendor/autoload.php';
```
Comment (clue) [inline tests/bootstrap.php]: This LGTM (IMO) :+1: 

Note however that removing this condition means we can no longer run the tests if this is installed as a composer dependency.

Personally, I would consider this a non-issue, but this has been raised before: https://github.com/reactphp/dns/pull/24#issuecomment-117041513

**Comment by clue** [comment]
LGTM :+1: 

Ping @cboden, @jsor

File: src/ArrayCache.php [hunk 3]
```diff
@@ -6,8 +6,14 @@
 
 class ArrayCache implements CacheInterface
 {
-    private $data = array();
+    /**
+     * @var array
+     */
+    protected $data = array();
```
Comment (jsor) [inline src/ArrayCache.php]: Whats the reason to change this from `private` to `protected`?
